src/Microsoft.Health.Fhir.Core/Features/Context/IFhirRequestContext.cs

```diff
@@ -19,5 +19,10 @@ public interface IFhirRequestContext : IRequestContext
         IList<OperationOutcomeIssue> BundleIssues { get; }
 
         bool IncludePartiallyIndexedSearchParams { get; set; }
+
+        /// <summary>
+        /// Indicates whether this request is part of the execution of a batch or transaction request.
+        /// </summary>
+        bool ExecutingBatchOrTransaction { get; set; }
```

Comment:
@Ivanidzo4ka can this now replace [IProvenanceHeaderState.cs](https://github.com/microsoft/fhir-server/blob/main/src/Microsoft.Health.Fhir.Shared.Api/Features/Resources/IProvenanceHeaderState.cs)?